src/node_buffer.cc

```diff
@@ -1182,8 +1207,12 @@ void Swap16(const FunctionCallbackInfo<Value>& args) {
   THROW_AND_RETURN_UNLESS_BUFFER(env, args.This());
   SPREAD_ARG(args.This(), ts_obj);
 
-  for (size_t i = 0; i < ts_obj_length; i += 2) {
-    SWAP_BYTES(ts_obj_data, i, i + 1);
+  CHECK_EQ(ts_obj_length % 2, 0);
+  uint16_t* data16 = reinterpret_cast<uint16_t*>(ts_obj_data);
```

Comment:
> I'm not super keen on adding something that will improve perf ~1/2 to 1/8th of the time

That's not totally fair. IIRC one main reason for this operation was to flip the bits on a file. Which would probably be performed the moment after the file is received. Meaning it'll always be aligned.

>  if I were to add a byte-wise fall-through to handle unaligned buffers.

This is exactly what I was hoping for.

> Sorry for not catching this prior to submitting the PR!

No worries. I only know about this because I submitted a PR last year that relied on the same technique a year ago.

IMO keep everything you have and add the code for fallback if it's unaligned. That should still give a good performance increase over what's currently there.